rasterio/rio/main.py

```diff
@@ -75,3 +74,6 @@ def main_group(ctx, verbose, quiet, aws_profile, gdal_version):
     ctx.obj = {}
     ctx.obj['verbosity'] = verbosity
     ctx.obj['aws_profile'] = aws_profile
+    env = rasterio.Env(CPL_DEBUG=(verbosity > 2), profile_name=aws_profile)
+    env.get_aws_credentials()
+    ctx.obj['env'] = env
```

Comment:
@brendan-ward @perrygeo I'm not sure why I didn't think of this before, but `main_group` is a great place to configure our environment, yes?